### PR TITLE
Score cone high auto

### PR DESCRIPTION
**Pull request by jherrmann06**

### CODE REVIEW

File: src/main/java/frc/robot/commands/groups/ExtendAndScoreCone.java [hunk 1]
```diff
@@ -22,16 +21,12 @@ public ExtendAndScoreCone(DrivetrainSubsystem drive, FloorArmSubsystem floorArm,
       new WaitCommand(0.5),
       new FloorMoveArmToPostion(floorArm, Constants.FLOOR_ARM_INSIDE_ROBOT_POSITION),
       new MoveArmToPosition(arm, Constants.ARM_SCORE_HIGH_POSITION),
-      //new WaitUntilCommand(() -> Math.abs(arm.getPosition() - arm.getTargetPosition()) < Constants.ARM_ENCODER_COUNT_ERROR),
       new WaitCommand(1),
-      new SetDriveTranslationTarget(drive, getTranslationTarget(drive, 1)),
-      new DropCone(intake).withTimeout(1),//could probably tune this down
-      new SetDriveTranslationTarget(drive, getTranslationTarget(drive, -1)),
+      new RepeatCommand(new InstantCommand(() -> drive.drive(-1, 0, 0), drive)).withTimeout(0.5),
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/groups/ExtendAndScoreCone.java]: image

I think something like the StartEndCommand would be ideal here, so that you can set the drive to 0

File: src/main/java/frc/robot/commands/groups/ExtendAndScoreCone.java [hunk 2]
```diff
@@ -22,16 +21,12 @@ public ExtendAndScoreCone(DrivetrainSubsystem drive, FloorArmSubsystem floorArm,
...
+      new DropCone(intake).withTimeout(0.5),
+      new RepeatCommand(new InstantCommand(() -> drive.drive(1, 0, 0), drive)).withTimeout(0.5),
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/groups/ExtendAndScoreCone.java]: Same here

File: src/main/java/frc/robot/commands/groups/UpdatePoseParallel.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import frc.robot.subsystems.DrivetrainSubsystem;
+import frc.robot.subsystems.Vision;
+
+public class UpdatePoseParallel extends ParallelDeadlineGroup {
+
+  public UpdatePoseParallel(Command deadlineCommand, Vision vision, DrivetrainSubsystem drivetrainSubsystem) {
+    super(deadlineCommand);
+    addCommands(new VisionPoseUpdate(vision, drivetrainSubsystem));
+  }
+}
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/groups/UpdatePoseParallel.java]: This change doesnt appear to be related to scoring high. Could it be deferred to another PR or removed?

File: src/main/java/frc/robot/util/AutoGenerator.java [hunk 4]
```diff
@@ -77,9 +79,21 @@ private StartLocation(int value) {
         }
     }
 
-    public AutoGenerator(DrivetrainSubsystem drivetrainSubsystem, FloorArmSubsystem floorArmSubsystem, ArmSubsystem armSubsystem, Intake intake, FloorIntake floorIntake) {   
+    public AutoGenerator(DrivetrainSubsystem drivetrainSubsystem, FloorArmSubsystem floorArmSubsystem, ArmSubsystem armSubsystem, Intake intake, FloorIntake floorIntake, Vision vision) {
```
Comment (jdlanning) [inline src/main/java/frc/robot/util/AutoGenerator.java]: Same with this vision subsystem input

File: src/main/java/frc/robot/commands/groups/UpdatePoseParallel.java (hunk 3 above)
Comment (jherrmann06) [reply src/main/java/frc/robot/commands/groups/UpdatePoseParallel.java]: Comitted that by mistake, removed

File: src/main/java/frc/robot/util/AutoGenerator.java [hunk 5]
```diff
@@ -83,6 +84,18 @@ private StartLocation(int value) {
...
+        locationSelector.addOption("Area Three", StartLocation.THREE);
+
+        pathActionSelector.setDefaultOption("Score game piece and exit", PathActions.SCORE_AND_EXIT);
+        pathActionSelector.addOption("Score cone high, exit community, and balance", PathActions.SCORE_EXIT_BALANCE);
+        pathActionSelector.addOption("Score game piece, rotate and exit", PathActions.SCORE_ROTATE_AND_EXIT);
+
+        startActionSelector.setDefaultOption("Score cone low", StartActions.SCORE_CONE_LOW);
+        startActionSelector.addOption("Score cube low", StartActions.SCORE_CUBE_LOW);
+        startActionSelector.addOption("Score cone high", StartActions.SCORE_CONE_HIGH);
+        startActionSelector.addOption("Score cube high", StartActions.SCORE_CUBE_HIGH);
```
Comment (jdlanning) [inline src/main/java/frc/robot/util/AutoGenerator.java]: I think these are already defined below

File: src/main/java/frc/robot/commands/groups/ExtendAndScoreCone.java [hunk 6]
```diff
@@ -22,16 +23,20 @@ public ExtendAndScoreCone(DrivetrainSubsystem drive, FloorArmSubsystem floorArm,
...
+      driveDistance(drive, 0.5, 1).withTimeout(0.5),
       new MoveArmToPosition(arm, Constants.ARM_INSIDE_ROBOT_POSITION)
     );
   }
 
-  private Translation2d getTranslationTarget(DrivetrainSubsystem drive, double dx) {
-    return drive.getPose().getTranslation().plus(new Translation2d(dx, 0));
+  private Command driveDistance(DrivetrainSubsystem drive, double distance, double velocity) {
+    return new StartEndCommand(
+      () -> new RepeatCommand(new InstantCommand(() ->drive.drive(velocity, 0, 0), drive)),
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/groups/ExtendAndScoreCone.java]: This should be able to just be a drive method call instead of needing the repeat command. Since the chassis speed persist this command will set them when it’s initialized and set them to 0 when it ends. Since the speeds are constant we don’t need to mimic the execute method with a repeat command

File: src/main/java/frc/robot/RobotContainer.java [hunk 7]
```diff
@@ -178,7 +177,7 @@ private void configureBindings() {
         ));
         operatorControllerRightBumper.whileTrue(
           new ConditionalCommand(
-            new IntakeCube(intake),
+            new IntakeCone(intake),
```
Comment (jdlanning) [inline src/main/java/frc/robot/RobotContainer.java:184]: This seems like an accident?

File: src/main/java/frc/robot/util/AutoGenerator.java (hunk 5 above)
Comment (jdlanning) [reply src/main/java/frc/robot/util/AutoGenerator.java]: Still needs to be looked at

File: src/main/java/frc/robot/RobotContainer.java (hunk 7 above)
Comment (jherrmann06) [reply src/main/java/frc/robot/RobotContainer.java:184]: No, that was a deliberate change. We never use the end of arm cube intake, and it is easier for me to use the same button to intake everything. I would like to keep this change.

**Review by jdlanning — APPROVED** [review]
